osu.Game/Database/RealmArchiveModelImporter.cs

```diff
@@ -179,6 +179,30 @@ public async Task<IEnumerable<Live<TModel>>> Import(ProgressNotification notific
 
         public virtual Task<Live<TModel>?> ImportAsUpdate(ProgressNotification notification, ImportTask task, TModel original) => throw new NotImplementedException();
 
+        public async Task<ExternalEditOperation<TModel>> BeginExternalEditing(TModel model)
+        {
+            string mountedPath = Path.Join(Path.GetTempPath(), model.Hash);
+
+            if (Directory.Exists(mountedPath))
+                Directory.Delete(mountedPath, true);
+
+            Directory.CreateDirectory(mountedPath);
+
+            foreach (var realmFile in model.Files)
+            {
+                string sourcePath = Files.Storage.GetFullPath(realmFile.File.GetStoragePath());
+                string destinationPath = Path.Join(mountedPath, realmFile.Filename);
+
+                Directory.CreateDirectory(Path.GetDirectoryName(destinationPath)!);
+
+                using (var inStream = Files.Storage.GetStream(sourcePath))
+                using (var outStream = File.Create(destinationPath))
+                    await inStream.CopyToAsync(outStream).ConfigureAwait(false);
+            }
+
+            return new ExternalEditOperation<TModel>(this, model, mountedPath);
```

Comment:
Q: 
Would it be better to have fixed mounted path? It will let user easier to load the project like vscode.
Also, Lazer can only edit one beatmap one time, so there's no need to give the folder a hash?

Reply:
Except filesystems are not as durable as you think they are and reusing a path leads to many issues like "oh this folder isn't writable because someone broke something" "oh this directory is being used by something else and therefore cannot be written to".

I agree with the choice of not using a fixed path.

Reply:
Hmm... I'll have a think about this one. I can see the benefits for having it fixed, but as @bdach touches on it would need extra error handling.

Reply:
I think this should be considered as a follow-up effort if ever.

If we're to allow this, we'd probably not want to delete the files when re-importing so the user can keep an external editor open or something. It has a lot of considerations and a lot of opportunity for breakage.

Would probably say that it's a lower priority and time is best spent elsewhere now that we have the initial support in, unless we get a huge number of complaints from users that what has been implemented doesn't meet their needs.